.github/workflows/ci-cd-pipeline.yml

```diff
@@ -1,88 +1,74 @@
 name: CI-CD Pipeline
 
 on:
+  push:
+    branches: [test-branch]
   pull_request:
-    branches:
-      - main
+    branches: [main]
   schedule:
-    - cron: "0 0 * * *"  # Запуск по расписанию (каждый день в 00:00)
+    - cron: "0 0 * * *"
   workflow_dispatch:
     inputs:
       browser:
         description: 'Select browser for UI tests'
         required: true
         default: 'chrome'
         type: choice
-        options:
-          - chrome
-          - firefox
-          - edge
+        options: [chrome, firefox, edge]
 
 jobs:
   api-tests:
     runs-on: ubuntu-latest
-    name: Run API Tests
+    name: API Tests
     steps:
-      - name: Checkout API repository
-        uses: actions/checkout@v2
+      - name: Checkout API repo
+        uses: actions/checkout@v4
         with:
-          repository: 'AlibekNamazbayev/APIAutomation'
+          repository: AlibekNamazbayev/APIAutomation
           token: ${{ secrets.GITHUB_TOKEN }}
-        
-      - name: Setup .NET
-        uses: actions/setup-dotnet@v3
+          path: api-repo
+
+      - name: Setup .NET 9
+        uses: actions/setup-dotnet@v4
         with:
-          dotnet-version: '9.0.x'
+          dotnet-version: | 
+            9.0.x
```

Comment:
why 2 versions? the versions are backward compatible, so 9 will execute 8.
9 is enough.
Aslo I would suggest to move it to the variables of github actions, in that case u dont need to update the code of workflow, just update the variable value